### ISSUE TITLE
Home Assistant Temporary Workaround

### ISSUE THREAD

**Issue by Supreme500** [derkalle4/python3-idotmatrix-client#30]
Hi!

I wanted to first start off by thanking you for starting this project. I have gotten Home Assistant to successfully send commands to the Pixel Screen (I have a 32X32). I use a HACS repo called  SSHCommand which can be found here AlexxIT/SSHCommand

Using that allows you to perform automations by calling a Service and selecting SSHCommand to send the commands to the remote Linux client (in my case OrangePI).

If you need more info please do not hesitate to contact me. Would be nice to have my Linux client always connected so that when an automation gets triggered we don't see the Bluetooth connection logo every time on the Pixel Display. Also, as a quick side note, not all GIFs seem to work with the commands **--set-gif** and **--process-gif**. I have tried many GIFs of different resolutions and file sizes. If I upload the same GIF using their phone APP it works. Seems like some incompatibility or the unit is picky with what GIFs are sent using your library.

Thanks

**Comment by tigers75** [comment]
Could you care to elaborate more? If I understand this correctly you have a separate Raspberry with this repo installed and use the SSHCommand to send commands to it? How do you send the gifs? Are they stored on the HA machine or on the remote Raspberry?
Do you think it could work with a BT proxy with ESP32 installed, sending data directly from the HA machine?

**Comment by Supreme500** [comment]
> Could you care to elaborate more? If I understand this correctly you have a separate Raspberry with this repo installed and use the SSHCommand to send commands to it? How do you send the gifs? Are they stored on the HA machine or on the remote Raspberry?
> 
> Do you think it could work with a BT proxy with ESP32 installed, sending data directly from the HA machine?

Correct I have a seperate linux device (OrangePi) that has all the GIFs and images stored onto it. SSHCommand allows me to send Linux commands to the OrangePi (or any Linux device), which is where I send the GIF command that was developped from this project. As per ESP32, I am not sure as I do not have a testbed for that.

Thanks

**Comment by derkalle4** [comment]
Hi @Supreme500,

many thanks for your interest in my project! If you are familiar with Python you can simply use my library instead of this client to directly utilize this in a small homeassistant script. A homeassistent integration is planned but unfortunately I haven't got time to implement it properly: https://github.com/derkalle4/python3-idotmatrix-library/

Kind Regards,
Kalle

**Comment by aptonline** [comment]
This is great, I came here looking for an API and Home Assistant integration and found this post. @Supreme500 have you managed to get the library working with HA directly.  Happy to help support/test any work on this.

**Comment by Supreme500** [comment]
> This is great, I came here looking for an API and Home Assistant integration and found this post. @Supreme500 have you managed to get the library working with HA directly.  Happy to help support/test any work on this.

Hi! I have not looked into this yet as I have been very busy lately. Let me know if you get it working by chance. Thanks

**Comment by Dokport** [comment]
Thanks for suggesting this, I have it up and running on a Pi Zero W, and it works. 
How fast response are you getting? I'm seeing almost 9 seconds for a simple turn off screen command, but I suspect its partly due to the slow processing of the Pi Zero?
I tried to install the library directly in Home Assistant, but I didn't manage to get it running, and ChatGPT started running in circles on me

**Comment by julianjwong** [comment]
> Thanks for suggesting this, I have it up and running on a Pi Zero W, and it works. How fast response are you getting? I'm seeing almost 9 seconds for a simple turn off screen command, but I suspect its partly due to the slow processing of the Pi Zero? I tried to install the library directly in Home Assistant using AppDaemon Add-on, but I didn't manage to get it running, and ChatGPT started running in circles on me

So is your workaround to install the client on a separate PiZW and call it in the same way @Supreme500 does it?

I installed the client in HA and can run via the terminal but can't use shell commands to get the commands running. Based on my research looks like this is because HA doesn't allow importing of external libraries during script execution? Have tried to work out if I can hack this through without luck. AppDaemon is beyond my skillset I think.

**Comment by derkalle4** [comment]
We should definitely add a Homeassistent integration client separated from this client. The library is already separated for this exact use case. I may can create a repository with a starting point but do not have the time due to Postcovid to work on this actively.

**Comment by julianjwong** [comment]
I started a thread on the Home Assistant community. Making some good progress using pyscript and the idotmatrix library. I'm learning as I go with the help of others but will share on the HA forums here: https://community.home-assistant.io/t/help-getting-idotmatrix-working-in-ha/829643

Sending text to the 16x16 isn't working though via the library or the client. I'm also noticing that the device drops bluetooth broadcasting sometimes. Sometimes it resolves itself and becomes discoverable via scan and other times I have to power cycle it.

**Comment by derkalle4** [comment]
Yeah, either the device and/or my Bluetooth implementation is wrong somewhere. At least for the 16x16 images I still have to fix something if I remember correctly.

Thanks for your work ;)
